File: javascript/medium/containerWithMostWater.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let max = 0, i = 0, j = height.length-1;
    while(i < j) {
        const area = Math.min(height[i], height[j]) * (j - i);
        max = Math.max(max, area);
        if(height[i] < height[j]) {
            i++;
        } else {
            j--;
        }
    }
    return max;
};

File: ruby/medium/container_with_most_water.rb
# @param {Integer[]} height
# @return {Integer}
def max_area(height)
    max_area, i, j = 0, 0, height.length-1
    while i < j
        area = [height[i], height[j]].min * (j-i)
        max_area = [max_area, area].max
        if height[i] < height[j]
            i += 1
        else
            j -= 1
        end
    end
    return max_area
end
